public/app/features/query/state/DashboardQueryRunner/UnifiedAlertStatesWorker.ts

```diff
@@ -29,6 +31,10 @@ export class UnifiedAlertStatesWorker implements DashboardQueryRunnerWorker {
       return false;
     }
 
+    if (!contextSrv.hasPermission(AccessControlAction.AlertingRuleRead)) {
```

Comment:
No need to check `AccessControlAction.AletingRuleExternalRead`?

Reply:
A further question here - what's the granularity for `AletingRuleExternalRead` action? Can it be scoped per dashboard? If so, you'll probably want to check whether the user has been granted `AletingRuleExternalRead` action for this particular dashboard (we've done similar things with [teams](https://github.com/grafana/grafana/blob/main/public/app/features/teams/TeamList.tsx#L76-L78), data sources etc).

Reply:
I added check for `AccessControlAction.AletingRuleExternalRead`
`AccessControlAction.AletingRuleExternalRead` has a scope of `datasource:*` so if the user doesn't have access to a data source then they won't see alerts from that data source
The `UnifiedAlertStateWorker` is only used to show the alerting status indicator for each panel with alerts assigned. If the `canWork` method returns false then the indicator is not displayed